### PR TITLE
Add Pneumatics to the Registrar. Fixes #33

### LINKED ISSUE #33: Registrar - Pneumatics
Solenoid Objects can not be stored by the Registrar currently. There should be extra methods added to remedy this issue

### PR DESCRIPTION
**Pull request by Philanatidae**
In reference to issue #33.

### CODE REVIEW

File: src/main/java/jaci/openrio/toast/lib/registry/Registrar.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
      */
     public synchronized  <T extends Type> T fetch(ID id, Class<T> clazz, Supplier<T> creator) {
         Type in_register = registered.get(id);
-        if (in_register == null) {
+        if (in_register.equals(null)) {
```
Comment (JaciBrunning) [inline src/main/java/jaci/openrio/toast/lib/registry/Registrar.java]: null.equals(null) will cause a nullpointer exception. Use the original `in_register == null` instead

**Comment by Philanatidae** [comment]
@JacisNonsense Commit 4f218ab fixes the null pointer exception possibility. I was speed reading the code and made a mistake. It also adds a check to SolenoidID for a null pointer exception. Commit 5d968a0 ammends how the registry checks for existing IDs (since checking hashes for keys won't work with creating new instances of SolenoidID class).

File: src/main/java/jaci/openrio/toast/lib/registry/Registrar.java [hunk 2]
```diff
@@ -32,6 +32,15 @@
...
+
+        if(in_register == null) {
+            for (ID key : registered.keySet()) {
+                if (key.equals(id)) {
+                    in_register = registered.get(key);
+                }
+            }
+        }
+
         if (in_register == null) {
```
Comment (JaciBrunning) [inline src/main/java/jaci/openrio/toast/lib/registry/Registrar.java:44]: Could this be combined with the if() statement above it? Both are checking for a null

Comment (Philanatidae) [inline src/main/java/jaci/openrio/toast/lib/registry/Registrar.java:44]: They have different functions. I suppose it _could_ be combined, but the whole fetch method would have to be restructured. Think of this commit's code as an expanded version of line 34. If the hash comparison failed, we run a comparison against values. If that fails, we move on and create a new instance of the object.

Comment (JaciBrunning) [inline src/main/java/jaci/openrio/toast/lib/registry/Registrar.java:44]: Ahh, I see. Gotcha. I'll give this a test tonight after merging

**Pull request merged**